src/Polly.Core/Strategy/OutcomePredicate.TResult.cs

```diff
@@ -0,0 +1,222 @@
+using System;
+using Polly.Strategy;
+
+namespace Polly.Strategy;
+
+/// <summary>
+/// Predicate that uses <see cref="Outcome{TResult}"/> and <typeparamref name="TResult"/> as an input.
+/// </summary>
+/// <typeparam name="TArgs">The type of arguments the predicate uses.</typeparam>
+/// <typeparam name="TResult">The result type that this predicate handles.</typeparam>
+public sealed class OutcomePredicate<TArgs, TResult>
```

Comment:
Maybe `TOutcome`? Whatever the name I think the name should match the type parameters on things like `OutcomeGenerator` like with the previous comment.

Reply:
TResult is consistent with naming in `OutcomeGenerator`:

``` csharp
OutcomeGenerator<TArgs, TValue> SetGenerator<TResult>(Func<Outcome<TResult>, TArgs, TValue> generator)
```

We also use the same name at various places:

``` csharp
public readonly struct Outcome<TResult>

public class ResilienceStrategy
{
    public TResult Execute<TResult>(
        Func<CancellationToken, TResult> callback,
        CancellationToken cancellationToken = default)
}
```